pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -102,69 +101,84 @@ func (p RESTStorageProvider) GroupName() string {
 // PostStartHook returns the hook func that launches the config provider
 func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
 	return PostStartHookName, func(hookContext genericapiserver.PostStartHookContext) error {
-		flowcontrolClientSet := flowcontrolclient.NewForConfigOrDie(hookContext.LoopbackClientConfig)
+		clientset := flowcontrolclient.NewForConfigOrDie(hookContext.LoopbackClientConfig)
+
 		go func() {
-			const retryCreatingSuggestedSettingsInterval = time.Second
-			err := wait.PollImmediateUntil(
-				retryCreatingSuggestedSettingsInterval,
-				func() (bool, error) {
-					should, err := shouldEnsureSuggested(flowcontrolClientSet)
-					if err != nil {
-						klog.Errorf("failed getting exempt flow-schema, will retry later: %v", err)
-						return false, nil
-					}
-					if !should {
-						return true, nil
-					}
-					err = ensure(
-						flowcontrolClientSet,
-						flowcontrolbootstrap.SuggestedFlowSchemas,
-						flowcontrolbootstrap.SuggestedPriorityLevelConfigurations)
-					if err != nil {
-						klog.Errorf("failed ensuring suggested settings, will retry later: %v", err)
-						return false, nil
-					}
-					return true, nil
-				},
-				hookContext.StopCh)
-			if err != nil {
-				klog.ErrorS(err, "Ensuring suggested configuration failed")
-
-				// We should not attempt creation of mandatory objects if ensuring the suggested
-				// configuration resulted in an error.
-				// This only happens when the stop channel is closed.
-				// We rely on the presence of the "exempt" priority level configuration object in the cluster
-				// to indicate whether we should ensure suggested configuration.
-				return
-			}
+			ensureWithRetry(clientset, hookContext)
 
-			const retryCreatingMandatorySettingsInterval = time.Minute
-			_ = wait.PollImmediateUntil(
-				retryCreatingMandatorySettingsInterval,
+			// Now we stomp changes made to mandatory configuration
```

Comment:
This is no longer needed - it's part of ensureWithRetry, right?

Reply:
currently, the mandatory configuration poller runs indefinitely every `1m`. The idea is, after the initial creation process, we check every minute and stomp any changes.

This PR just separates these two concerns:
- initial creation of mandatory configuration, retry interval of `1s` like the other pollers inside `ensureWithRetry`
- after initial configuration looks good, poll every minute and stomp changes to mandatory configuration.

so the sequence this PR proposes is:
- A: ensure suggested configuration, with retry interval of `1s`
- B: ensure mandatory configuration, with retry interval of `1s`
- C: remove unwanted configuration, with retry interval of `1s`
- D: stomp changes to mandatory configuration, with interval of `1m`

Reply:
Hmm - I think I don't agree with it.

I think with auto-update feature, we should actually try to auto-update (or auto-remove etc.) suggested configuration periodically too (doing that just on kube-apiserver start seems a bit strange, bcause in case of issues the mitigation is to retart the kub-apiserver, that doesn't sound right to me).